### ISSUE TITLE
Node exit with error "Failed to iterate between blocks on diverging forks:"

### ISSUE THREAD

**Issue by vht210** [iron-fish/ironfish#880]
Update to 0.1.18
- Rung node for a while then it stopped with the error below (Reproduced with multiple  node/miners)

git/ironfish/ironfish/build/src/blockchain/blockchain.js:286
            throw new Error('Failed to iterate between blocks on diverging forks:' +
                  ^

Error: Failed to iterate between blocks on diverging forks: curr: 00000...24849, end: 00000...dd4cb, progress: 63094/63093
    at Blockchain.iterateToHashes (/home/tung/git/ironfish/ironfish/build/src/blockchain/blockchain.js:286:19)
    at iterateToHashes.next (<anonymous>)
    at Blockchain.iterateTo (/home/tung/git/ironfish/ironfish/build/src/blockchain/blockchain.js:265:26)
    at iterateTo.next (<anonymous>)
    at Blockchain.iterateAllTransactions (/home/tung/git/ironfish/ironfish/build/src/blockchain/blockchain.js:733:26)
    at async Accounts.scanTransactions (/home/tung/git/ironfish/ironfish/build/src/account/accounts.js:393:80)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by NullSoldier** [comment]
This issue is caused by two things
 * If you start a rescan using `ironfish accounts:rescan` when your wallet is on a fork
 * If you import an account using `ironfish accounts:import` when your wallet is on a fork

A temporary workaround is to clear your wallet stater using `ironfish accounts:rescan —reset`

**Comment by NullSoldier** [comment]
This issue is happening because of how we handle rescans. 

1. When a rescan occurs, the wallet will never update itself until the rescan completes.
2. When the node starts and a rescan has started it will always try to rescan first

This means that because the rescan is broken, the wallet is stuck in a broken state permanently. Let's now dive into why the rescan is broken.


```
G -> A1 -> A2
        -> B2 -> B3

Accounts Head: A2
Chain Head:    B3
```

When we do a rescan, we try to iterate from the genesis block (G) to the accounts head (A2), but the block chain has re-orged to B3, the newer heaviest chain. If the accounts head were to update, it would reorganize as well to B3. Since we're doing a re-org once a scan has started the accounts will wait for the rescan to finish before updating the accounts head.

Now the way we try to iterate from G -> A2 is that we use the heaviest chain iteration. That means we're going to iterate in this order: `G -> A1 -> B2` before realizing were not on target and fail with the error that's in the ticket. It's now unrecoverable because there's no way out of this since accounts is stuck at A2.

**Comment by deekerno** [comment]
Fixed by #937.
